c++/src/kj/common.h

```diff
@@ -1379,6 +1379,25 @@ class Maybe {
     }
   }
 
+  template <typename F, typename RetOk = EnableIf<isSameType<T, decltype(instance<F>()())>()>>
```

Comment:
Probably should use canConvert() instead of isSameType()?

Reply:
Is that testing implicit or explicit conversion? If the latter then there needs to be an additional T constructor to wrap the result, right?

Reply:
`canConvert()` only tests for implicit conversion, which is what we want here I think.